src/transformers/models/dpt/modeling_dpt.py

```diff
@@ -0,0 +1,1127 @@
+            [
+                nn.ConvTranspose2d(
+                    in_channels=out_channels[0], out_channels=out_channels[0], kernel_size=4, stride=4, padding=0
+                ),
+                nn.ConvTranspose2d(
+                    in_channels=out_channels[1], out_channels=out_channels[1], kernel_size=2, stride=2, padding=0
+                ),
+                nn.Identity(),
+                nn.Conv2d(
+                    in_channels=out_channels[3], out_channels=out_channels[3], kernel_size=3, stride=2, padding=1
+                ),
+            ]
+        )
+        if config.readout_type == "project":
+            self.readout_projects = nn.ModuleList()
+            for _ in range(len(self.projects)):
+                self.readout_projects.append(
+                    nn.Sequential(nn.Linear(2 * config.hidden_size, config.hidden_size), ACT2FN[config.hidden_act])
+                )
+
+    def forward(self, inputs):
+        """
+        Inputs: list of torch.FloatTensor, each of shape (B, L + 1, C).
+        """
+        assert isinstance(inputs, list)
+        out = []
+
+        for i, x in enumerate(inputs):
+            # reshape to (B, C, H, W)
+            x, cls_token = x[:, 1:], x[:, 0]
+            B, L, C = x.shape
+            size = int(math.sqrt(L))
+            x = x.reshape(B, size, size, C)
+            x = x.permute(0, 3, 1, 2).contiguous()
+
+            feature_shape = x.shape
+            if self.config.readout_type == "project":
+                x = x.flatten(2).permute((0, 2, 1))
+                readout = cls_token.unsqueeze(1).expand_as(x)
+                x = self.readout_projects[i](torch.cat((x, readout), -1))
+                x = x.permute(0, 2, 1).reshape(feature_shape)
+            elif self.config.readout_type == "add":
+                x = x.flatten(2) + cls_token.unsqueeze(-1)
+                x = x.reshape(feature_shape)
+            x = self.projects[i](x)
+            x = self.resize_layers[i](x)
+            out.append(x)
+
+        return out
+
+
+class DPTPreActResidualConvUnit(nn.Module):
+    """
+    ResidualConvUnit, pre-activate residual unit.
+
+    Args:
+        in_channels (int): number of channels in the input feature map.
+        act_cfg (dict): dictionary to construct and config activation layer.
+        norm_cfg (dict): dictionary to construct and config norm layer.
+        stride (int): stride of the first block. Default: 1
+        dilation (int): dilation rate for convs layers. Default: 1.
+        init_cfg (dict, optional): Initialization config dict. Default: None.
+    """
+
+    def __init__(self, config):
+        super().__init__()
+
+        self.use_batch_norm = config.use_batch_norm
+        self.act1 = nn.ReLU()
+        self.conv1 = nn.Conv2d(
+            config.channels,
+            config.channels,
+            kernel_size=3,
+            stride=1,
+            padding=1,
+            bias=not self.use_batch_norm,
+        )
```

Comment:
Since this is a `Layer`, I think you can ditch the `config` and just pass the parameters you need so it's way easier to read/understand/reuse in other models. Let me know what do you thinkg

Reply:
What do you mean here?

Reply:
I was asking if we should use the config here or pass everything we need from it as parameters. I am not sure if we have a rule in place for how deep should the config passed